### DISCUSSION TITLE
Known ordering for stitiching

### DISCUSSION THREAD

**Discussion by dariengit** [OpenStitching/stitching#138] [General]
Hi there,
Thanks for your effort, very good project to solve the stitching problem.

But Question here:
If I already know the ordering of small pieces with file names, like the example below, is it possible to give order parameters with CLI to make a more accurate stitching result?

![image001](https://github.com/OpenStitching/stitching/assets/37905163/1b6dfd09-82bd-4a0d-8ad2-b71b4ac6f17e)

Thanks and regards,
Darien

**Comment by lukasalexanderweber** [comment]
For a single row or column of images you can just pass in the images in the correct order and set `--range_width 1`

See e.g. https://github.com/OpenStitching/stitching/issues/14, https://github.com/OpenStitching/stitching/issues/56 and https://github.com/OpenStitching/stitching/blob/main/tests/test_range_width_matcher.py.

The help in the cli is not that accurate, you could consider improving it. 

In your case, you would have to consider that `0-2 - 1-0` have no matches, so your input must be `0-0 0-1 0-2 1-2 1-1 1-0 2-0 ...` (writing this feels a bit like playing snake xD)

However, I don't know for "multi-row" panoramas if the result would be as good, you need to experiment.

Feel free to report if you have interesting findings
